Lib/fontTools/ttLib/tables/otTables.py

```diff
@@ -1359,6 +1359,59 @@ def toXML(self, xmlWriter, font, attrs=None, name=None):
 		xmlWriter.endtag(tableName)
 		xmlWriter.newline()
 
+	_LEAVES = {
+	    PaintFormat.PaintSolid,
+	    PaintFormat.PaintLinearGradient,
+	    PaintFormat.PaintRadialGradient,
+	    PaintFormat.PaintSweepGradient,
+	}
+
+	_HAS_PAINT = {
+		PaintFormat.PaintGlyph,
+		PaintFormat.PaintTransform,
+		PaintFormat.PaintTranslate,
+		PaintFormat.PaintRotate,
+		PaintFormat.PaintSkew,
+	}
```

Comment:
I think we can write this in a way that Just Works as paints change by using the otdata. Imagine _isLeafPaint(paint) taht just looks for Paint-typed fields, and a _childPaints(paint) that yanks all the Paint-typed fields. PaintColrLayers would require custom logic but I think that will remain unique.

That would mean changes like adding more paints for variable/not-variable Just Work here.

Reply:
If we land https://github.com/fonttools/fonttools/compare/colr_to_from_dicts (not strictly necessary, just that my instinct is there is shared logic here) I think we can have a generalized version of this. I'm picturing `traverse(root, filter, callback)`. Callback should be allowed to return things to add to the frontier so things like PaintColrLayers can work. For our use filter would just be is-a ot.Paint and for all types except PaintColrLayers children would automatically be identified.

Reply:
> I think we can write this in a way that Just Works as paints change by using the otdata

I just did that in https://github.com/fonttools/fonttools/pull/2177/commits/08cbce077f9f5c7d6779d35c91ed5e664abbc213, let me know if that's what you had in mind.

Reply:
LGTM